app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt

```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
     }
 
     override fun getLastModified(): String {
-        return if (secondOperand.isNotEmpty()) {
-            secondOperand
-        } else if (operator.isNotEmpty()) {
-            operator
-        } else {
-            firstOperand
+        return when {
+            secondOperand.isNotEmpty() -> {
+                secondOperand
+            }
+            operator.isNotEmpty() -> {
+                operator
+            }
+            else -> {
+                firstOperand
+            }
+        }
+    }
+
+    private fun operandEnabled(operand: String): Boolean {
+        return (operand != EMPTY_STRING && operand != SUB)
+    }
+
+    private fun operationEnabled(): Boolean {
+        var operationEnabled = true
+        if (getError() == Error.ERROR_DIVISION_BY_ZERO || getError() == Error.ERROR_INCOMPLETE_OPERATION) {
+            operationEnabled = false
+        }
+        return operationEnabled
+    }
+
+    override fun getError(): Error {
+        if (!operandEnabled(firstOperand) || !operandEnabled(secondOperand)) {
+            return Error.ERROR_INCOMPLETE_OPERATION
+        }
+        if (operandEnabled(firstOperand) && (operator == DIV) && (secondOperand == ZERO_STRING)) {
+            return Error.ERROR_DIVISION_BY_ZERO
+        }
+        return Error.NONE
+    }
+
+    private fun makeOperation(): Double {
+        when (operator) {
+            ADD -> {
+                return firstOperand.toDouble() + secondOperand.toDouble()
+            }
+            SUB -> {
+                return firstOperand.toDouble() - secondOperand.toDouble()
+
+            }
+            MUL -> {
+                return firstOperand.toDouble() * secondOperand.toDouble()
+            }
+            DIV -> {
+                return firstOperand.toDouble() / secondOperand.toDouble()
+            }
+        }
+        return DEFAULT_RESULT
+    }
+
+    override fun getOperation(): String {
+        return "$firstOperand $operator $secondOperand"
```

Comment:
NICEE! same you can simplify 
override fun getOperation() = "$firstOperand $operator $secondOperand"

Reply:
Solved!